### ISSUE TITLE
FR: Support for CraftQL

### ISSUE THREAD

**Issue by LeeCampbellr** [verbb/navigation#94]
### Description
Would love to see support with CraftQL for those of us working on some Headless CraftCMS builds, having the ability to query the navigation plugin from graphQL would be an amazing addition!

**Comment by engram-design** [comment]
Certainly want to support that, and I'll see what's involved!

**Comment by jesuismaxime** [comment]
@engram-design any roadmap for this feature ?

**Comment by edolyne** [comment]
@engram-design Now that Craft 3.3 supports GraphQL natively, is there a possibility for this to be added?

**Comment by engram-design** [comment]
Yep, this will be getting added, just getting myself more familiar with GraphQL (in general), but count it as 2nd on the todo list for the next release.

**Comment by davebeesley** [comment]
Hey @engram-design 
Are you able to give an update on this feature?
I am using this plugin for a project which will be heavily reliant on GraphQL, and have just stumbled upon a lack of support.
I just need to know if we need to find an alternative/workaround

**Comment by engram-design** [comment]
@davebeesley Will be in the next release shortly!

**Comment by engram-design** [comment]
Added in [1.2.0](https://github.com/verbb/navigation/releases/tag/1.2.0)

**Comment by davebeesley** [comment]
Hi @engram-design 
Thanks for sorting this, I am trying it now.
Am I to expect a flat array back from the GraphQL API?
That is what I get back (which I presume is normal behaviour)
But when I select children and their titles etc, I get a server error

```
query mainNavigationQuery {
  craftcms {
    nodes(navHandle: "mainNavigation") {
      ... on CraftCMS_Node {
        id
        title
        children {
          title
          id
        }
      }
      level
    }
  }
}
```

```
{
  "errors": [
    {
      "message": "Internal server error",
      "locations": [
        {
          "line": 3,
          "column": 5
        }
      ],
      "path": [
        "craftcms",
        "nodes"
      ],
      "stack": [
        "Error: Internal server error",
        "    at new CombinedError (/Users/davebeesley/Code/[project redacted]/node_modules/graphql-tools/dist/stitching/errors.js:83:28)",
        "    at Object.checkResultAndHandleErrors (/Users/davebeesley/Code/[project redacted]/node_modules/graphql-tools/dist/stitching/errors.js:101:15)",
        "    at CheckResultAndHandleErrors.transformResult (/Users/davebeesley/Code/[project redacted]/node_modules/graphql-tools/dist/transforms/CheckResultAndHandleErrors.js:10:25)",
        "    at /Users/davebeesley/Code/[project redacted]/node_modules/graphql-tools/dist/transforms/transforms.js:19:54",
        "    at Array.reduce (<anonymous>)",
        "    at applyResultTransforms (/Users/davebeesley/Code/[project redacted]/node_modules/graphql-tools/dist/transforms/transforms.js:18:23)",
        "    at /Users/davebeesley/Code/[project redacted]/node_modules/graphql-tools/dist/stitching/delegateToSchema.js:102:50",
        "    at step (/Users/davebeesley/Code/[project redacted]/node_modules/graphql-tools/dist/stitching/delegateToSchema.js:40:23)",
        "    at Object.next (/Users/davebeesley/Code/[project redacted]/node_modules/graphql-tools/dist/stitching/delegateToSchema.js:21:53)",
        "    at fulfilled (/Users/davebeesley/Code/[project redacted]/node_modules/graphql-tools/dist/stitching/delegateToSchema.js:12:58)",
        "    at process._tickCallback (internal/process/next_tick.js:68:7)"
      ]
    }
  ],
  "data": {
    "craftcms": {
      "nodes": null
    }
  }
}
```

**Comment by davebeesley** [comment]
Just wanted to check everything is correct before I head down the road of writing a crawler in JS to generate a nested array from the data I got back using parent id's

**Comment by engram-design** [comment]
@davebeesley Fixed in [1.2.1](https://github.com/verbb/navigation/releases/tag/1.2.1)

**Comment by davebeesley** [comment]
Awesome man. That has fixed it. Cheers :)
